Node editor

LLS-lopos opened the pull request:
Node editor for creation tests from Blender.
3 nodes currently available:
- Init (Use `sudo` if it doesn't launch, choose an STK executable or the one from the machine, choose to launch without extensions, and choose to load a personal asset folder)
- CLI which is there to write commands as one would in a Terminal.
- Run to execute the command (Run to launch from within Blender, Popen for execution independent of Blender)

LLS-lopos committed:
minimal/base nodal editor for test artiste creation

LLS-lopos committed:
cli for classic command and init for define game used et minimal option

LLS-lopos committed:
node for running test

LLS-lopos committed:
clean

LLS-lopos committed:
minor correction

LLS-lopos commented:
<img width="1111" height="610" alt="image" src="https://github.com/user-attachments/assets/6898a192-9bd1-4166-b526-61b0d3381dcd" />

For example, I don't define an executable, so it's the one that's installed. Then I add an extra folder and run a demo test.

LLS-lopos committed:
text adjustement and translation

LLS-lopos commented:
@Alayan-stk-2

LLS-lopos committed:
best management of nodal interface updates

LLS-lopos committed:
Simplify author name in bl_info

LLS-lopos commented:
@RQWorldblender

Alayan-stk-2 commented:
I've been a bit busy, but I've seen your ping. I'll look at it when I can.

LLS-lopos committed:
Merge branch 'supertuxkart:master' into stk-origin

LLS-lopos committed:
update

LLS-lopos committed:
check box for executable run_game or system game

LLS-lopos committed:
menu reorganization

LLS-lopos committed:
menu reorganization

LLS-lopos commented:
<img width="507" height="575" alt="image" src="https://github.com/user-attachments/assets/d8a83238-ff8a-4322-ae43-c15c19eb75d8" />
<img width="791" height="546" alt="image" src="https://github.com/user-attachments/assets/52c7b752-6e4e-4aa1-9a85-be1ae5339da8" />
Replace and unify the Init and Run menus in the Operator menu.

LLS-lopos committed:
command sudo just for linux and mac

LLS-lopos committed:
update minimal info for io_artiste

LLS-lopos committed:
adjust language

LLS-lopos committed:
adjust init_stk

LLS-lopos committed:
Merge branch 'supertuxkart:master' into stk-origin

Alayan-stk-2 commented:
I tried to test the PR, but Blender is quite confusing when you are not familiar with it. I couldn't find your menus through F3 or by looking into the node editor. It wasn't clear to me either how to create the required nodes if creating them is required.

Please provide step-by-step instructions to test your PR.

LLS-lopos commented:
You need to activate addons -> node -> STK RUNNER, then it will be possible to select an editor "STK RUNNER"
The Init node must be used first (it can be replaced by CLI, which is a command line), and Go Run Test should always be used to launch the commands.

F3 functions perfectly

---

The default init process is equivalent to launching the "supertuxkart" command, just like in a terminal (Super user is not available on Windows; if needed, open Blender as administrator for the same result).

Custom executable for testing from a git repository or an extracted zip file

Go Test Run:
option Run exécute supertuxkart sur Blender (blender se fige)
option Popen lance un processeus séparer

LLS-lopos committed:
add try/except in stk Panel and add traceback for potential error

LLS-lopos committed:
using the fonction get_fcurves for fixe error export for items

LLS-lopos committed:
now exportImage copy all texture used in blender file

LLS-lopos committed:
fixed preference for blender 5.1

LLS-lopos committed:
fixed export Texture/Animation Blender 5.X

Kart:
- export texture fixed (no limited to folder project)
Track:
- export texture fixed (no limited to folder project)

Panel:
the poll function already manages the display conditions of the panel, removing unnecessary checks

fonction get_fcurves migrate to stk_track_utils

LLS-lopos committed:
Merge remote-tracking branch 'origin/stk-origin' into stk-origin

LLS-lopos committed:
clean
